js_modules/dagster-ui/packages/ui-core/src/asset-graph/sidebar/Node.tsx

```diff
@@ -0,0 +1,311 @@
+                      <Menu>
+                        <MenuItem
+                          icon="materialization"
+                          text={
+                            <Box flex={{direction: 'row', alignItems: 'center', gap: 4}}>
+                              <span>Materialize</span>
+                              {loading ? <Spinner purpose="body-text" /> : null}
+                            </Box>
+                          }
+                          onClick={async (e) => {
+                            await showSharedToaster({
+                              intent: 'primary',
+                              message: 'Initiating materialization',
+                              icon: 'materialization',
+                            });
+                            onClick([node.assetKey], e, false);
+                          }}
+                        />
+                        {upstream.length || downstream.length ? <MenuDivider /> : null}
+                        {upstream.length ? (
+                          <MenuItem
+                            text={`View parents (${upstream.length})`}
+                            icon="list"
+                            onClick={() => {
+                              setShowParents(true);
+                            }}
+                          />
+                        ) : null}
+                        {upstream.length ? (
+                          <MenuItem
+                            text="Show upstream graph"
+                            icon="arrow_back"
+                            onClick={showUpstreamGraph}
+                          />
+                        ) : null}
+                        {downstream.length ? (
+                          <MenuItem
+                            text="Show downstream graph"
+                            icon="arrow_forward"
+                            onClick={showDownstreamGraph}
+                          />
+                        ) : null}
+                      </Menu>
+                    }
+                    hoverOpenDelay={100}
+                    hoverCloseDelay={100}
+                    placement="right"
+                    shouldReturnFocusOnClose
+                  >
+                    <ExpandMore style={{cursor: 'pointer'}}>
+                      <Icon name="more_horiz" color={Colors.Gray500} />
+                    </ExpandMore>
+                  </Popover>
+                </div>
+              ) : null}
+            </GrayOnHoverBox>
+          </Box>
+        </BoxWrapper>
+      </Box>
+    </>
+  );
+};
+
+const UpstreamDownstreamDialog = ({
+  title,
+  assets,
+  isOpen,
+  close,
+  selectNode,
+}: {
+  title: string;
+  assets: string[];
+  isOpen: boolean;
+  close: () => void;
+  selectNode: (e: React.MouseEvent<any> | React.KeyboardEvent<any>, nodeId: string) => void;
+}) => {
+  return (
+    <Dialog isOpen={isOpen} onClose={close} title={title}>
+      <DialogBody>
+        <Menu>
```

Comment:
Dialog with a menu in it?

Reply:
I think it would be preferable to reuse the `AssetKeysDialog` and `VirtualizedItemListForDialog`

Reply:
I'll switch to that